broker-daemon/block-order-worker/index.js

```diff
@@ -573,7 +574,15 @@ class BlockOrderWorker extends EventEmitter {
     // reject the entire block order if an underlying order fails
     osm.once('reject', async () => {
       try {
-        await this.failBlockOrder(blockOrder.id, osm.order.error)
+        if (osm.shouldRetry()) {
+          const validation = async () => {
+            await this.relayer.adminService.healthCheck({})
+            await this.workBlockOrder(blockOrder, Big(osm.order.baseAmount))
+          }
+          await exponentialBackoff(validation, 10)
```

Comment:
don't you want to retry the `shouldRetry` check as well?

Reply:
does `workBlockOrder` check that this block order hasn't been cancelled or failed?

Reply:
We should be okay kicking things off with the `shouldRetry` without having to check if it should retry each iteration. The broker would have reestablished a gRPC stream with the relayer and would have replaced the block order if the order is still active.

`workBlockOrder` does check if the block order is in a workable state (i.e. if the order is active).